core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -126,11 +126,11 @@
      * A setting describing the maximum number of characters that will be analyzed for a highlight request.
      * This setting is only applicable when highlighting is requested on a text that was indexed without
      * offsets or term vectors.
-     * The default maximum of 10000 characters is defensive as for highlighting larger texts,
-     * indexing with offsets or term vectors is recommended.
+     * This setting is defensive as for highlighting larger texts, indexing with offsets or term vectors is recommended.
+     * For 6.x the default value is not set or equals to -1.
      */
     public static final Setting<Integer> MAX_ANALYZED_OFFSET_SETTING =
-        Setting.intSetting("index.highlight.max_analyzed_offset", 10000, 1, Property.Dynamic, Property.IndexScope);
+        Setting.intSetting("index.highlight.max_analyzed_offset", -1, -1, Property.Dynamic, Property.IndexScope);
```

Comment:
I think we should have a custom validator to throw an exception when the value is set to `0`. Otherwise this would fail the upgrade to 7 where `0` is forbidden.